src/screens/StudentGroups.tsx

```diff
@@ -1,66 +1,61 @@
+import StudentGroupPreview from 'components/StudentGroupPreview'
+import { IStudentGroup } from 'interfacesAndTypes/interfacesAndTypes'
 
 const StudentGroups: React.FC = () => {
-  const { handleSubmit, errors, register, formState } = useForm<FormData>()
+  const [studentGroupNameInput, setStudentGroupNameInput] = React.useState('')
 
   const { data: user } = useUser()
 
-  const studentGroupsRef = useFirestore().collection('teachers').doc(user.uid).collection('studentGroups')
-  const studentGroupsData = useFirestoreCollectionData(studentGroupsRef)
-  console.log(studentGroupsData)
-
-  type FormData = {
-    studentGroupName: string
-  }
+  const history = useHistory()
 
-  //Steve is the druel-ist.
-  interface IClassroom {
-    studentGroupName: string
-    students: []
-    docId: string
-  }
-
-  const studentGroupRef = useFirestore().collection('studentGroups')
-  const studentGroupDocuments = useFirestoreCollectionData<IClassroom & { docId: string }>(studentGroupRef, {
+  const studentGroupsRef = useFirestore().collection('teachers').doc(user.uid).collection('studentGroups')
+  const studentGroupsDocuments = useFirestoreCollectionData<IStudentGroup & { docId: string }>(studentGroupsRef, {
     idField: 'docId',
   })
 
-  async function onSubmit(data: FormData, e: React.BaseSyntheticEvent | undefined) {
-    await studentGroupRef.add({
-      studentGroupName: data.studentGroupName,
-      students: [],
-    })
-    e?.target.reset()
+  async function submitHandler(e: React.BaseSyntheticEvent | undefined) {
+    e?.preventDefault()
+    try {
+      const result = await studentGroupsRef.add({
+        studentGroupName: studentGroupNameInput,
+        students: [],
+      })
+      history.push('/student-group/' + result.id)
+    } catch (err) {
+      console.log(err)
+    }
+    setStudentGroupNameInput('')
   }
 
   return (
     <>
-      <form onSubmit={handleSubmit(onSubmit)}>
-        <FormControl isInvalid={!!errors.studentGroupName}>
-          <FormLabel htmlFor="studentGroupName">Group Name</FormLabel>
-          <Input name="studentGroupName" placeholder="Group name" ref={register({ required: true, minLength: 3 })} />
-          {errors.studentGroupName && errors.studentGroupName.type === 'required' && (
-            <FormErrorMessage>This field is required</FormErrorMessage>
-          )}
-          {errors.studentGroupName && errors.studentGroupName.type === 'minLength' && (
-            <FormErrorMessage>Please enter a group name with at least 3 characters</FormErrorMessage>
-          )}
-        </FormControl>
-        <Button mt={4} colorScheme="teal" isLoading={formState.isSubmitting} type="submit">
-          Submit
-        </Button>
-      </form>
-      {studentGroupDocuments.data?.map(doc => {
+      <Box>
+        <FormLabel htmlFor="studentGroupName">Student Group Name</FormLabel>
+        <Input
+          onChange={e => setStudentGroupNameInput(e.target.value)}
```

Comment:
same note as above

Reply:
I think I should do it here since I'm passing in e.target.value to a setState call, right?  I think the other two I can just pass the reference.

Reply:
yeah debatable in this case i guess, you'd have to create a new function, go with your preference